### ISSUE TITLE
Image duplicates when tags change for an image

### ISSUE THREAD

**Issue by Sunako** [xeron/danbooru-ruby-grabber#6]
As said in the title, it seems like your script will download an image again if the tags for it changes, rather than just updating it. I observed this with yande.re only, so I've no idea if it affects other booru sites. I am using the storage dir option if that matters. It also makes a dupe symlink and entry in the files.bbs file.

Just an example image I found that had undergone a tag change and then got downloaded again: http://privatepaste.com/8c3ec5d562

**Comment by xeron** [comment]
https://github.com/xeron/danbooru-ruby-grabber/blob/9e52ce4737df6a9fc70954ad1b961376ca0af29a/lib/booru/posts.rb#L104

I always use filename from from file url, so if it changes script will download file again. Yeah, that affects all boards which use tags in filenames (danbooru itself doesn't, it uses MD5).

Possible solutions:
* Good one: Use md5 (bad for sorting by name) or post id (good for sorting) as a filename
* Bad one: Scan all filenames for post id (regex?)

**Comment by Sunako** [comment]
Personally I think scanning the files for the post ID isn't such a bad idea, in theory in should work. If you found a file with the same post ID, just overwrite the filename with whatever it has changed to and remake the symlink and entry in the files.bbs file. Maybe I'm wrong though, I don't know much about ruby so I've no idea if this is difficult.

**Comment by xeron** [comment]
It's not hard at all. The main concerns:

* Format can change
* Could be slow if you have a lot of files

Post id is much more reliable. May be it worth to add an option to choose filename pattern.

```
-f, --filename      Filename pattern. Supported options: id (default), md5, tags, url (old default)
```

**Comment by Sunako** [comment]
That sounds good, just let people decide what they want to do.

**Comment by xeron** [comment]
I've implemented this but please note that `-f id` is default now. It will download all images again unless you add `-f url`. But you could try to rename all files to post_id.ext format and remove files.bbs before running new version of script.

**Comment by Sunako** [comment]
Is it supposed to update the tags in the files.bbs file when you use the id option? It doesn't seem to if it is.

**Comment by xeron** [comment]
No, it's not. I've fixed an issue (a2efe88811f6daa89599cd0ef491d0e127d17ac8) which was preventing some files being added to files.bbs so it could add new records now, but I recommend to remove file completely to get consistent information.

BTW it still won't update tags if they change, but at least it won't download a file again. Changing lines in the middle of text file isn't simple task actually.

May be it worth to replace this files.bbs with sqlite database :) Using sqlite it'll be able to store and update tags and any other information about image (including resolution, author, etc etc etc).

**Comment by Sunako** [comment]
Using an sqlite DB could be a good idea. Also, I didn't think about deleting the files.bbs. I could just make a cronjob to delete and update it every now and then for now.

Thanks for your cool script btw.
